src/bin/platforms/freebsd.ts

```diff
@@ -0,0 +1,312 @@
+   */
+  private async checkUser() {
+    try {
+      // check if user exists
+      child_process.execSync(`id ${this.hbService.asUser} 2> /dev/null`);
+    } catch (e) {
+      // if not create the user
+      child_process.execSync(`pw useradd -q -n ${this.hbService.asUser} -s /usr/sbin/nologin 2> /dev/null`);
+      this.hbService.logger(`Created service user: ${this.hbService.asUser}`, 'info');
+    }
+  }
+
+  // TODO: shutdown doesn't exist in jails. Require jailutils, sysctl security.jail.jailed to check, jkill to restart
+  /**
+   * Allows the homebridge user to shutdown and restart the server from the UI
+   * There is no need for full sudo access when running using hb-service
+   */
+  private setupSudo() {
+    try {
+      const npmPath = child_process.execSync('which npm').toString('utf8').trim();
+      const shutdownPath = child_process.execSync('which shutdown').toString('utf8').trim();
+      const sudoersEntry = `${this.hbService.asUser}    ALL=(ALL) NOPASSWD:SETENV: ${shutdownPath}, ${npmPath}, /usr/local/bin/npm, /usr/sbin/service, /usr/bin/tail`;
+    
+      // check if the sudoers file already contains the entry
+      const sudoers = fs.readFileSync('/usr/local/etc/sudoers', 'utf-8');
+      if (sudoers.includes(sudoersEntry)) {
+        return;
+      }
+    
+      // grant the user restricted sudo privileges to /sbin/shutdown
+      child_process.execSync(`echo '${sudoersEntry}' | sudo EDITOR='tee -a' visudo`);
+    } catch (e) {
+      this.hbService.logger('WARNING: Failed to setup /etc/sudoers, you may not be able to shutdown/restart your server from the Homebridge UI.', 'warn');
+    }
+  }
+
+  /**
+   * Update Node.js
+   */
+  public async updateNodejs(job: { target: string; rebuild: boolean }) {
+    this.hbService.logger('Update Node.js using pkg manually.', 'fail');
+      process.exit(1);
+  }
+
+  /**
+   * Create the rc service script
+   */
+  private async createRCService() {
+    const nodePath = child_process.execSync('which node').toString('utf8').trim();
+    const rcFileContents = [
+      '#!/bin/sh',
+      '#',
+      '# PROVIDE: ' + this.rcServiceName,
+      '# REQUIRE: NETWORKING SYSLOG',
+      '# KEYWORD: shutdown',
+      '#',
+      '# Add the following lines to /etc/rc.conf to enable ' + this.rcServiceName + ':',
+      '#',
+      '#' + this.rcServiceName + '_enable="YES"',
+      '',
+      '. /etc/rc.subr',
+      '',
+      'name="' + this.rcServiceName + '"',
+      'rcvar="' + this.rcServiceName + '_enable"',
+      '',
+      'load_rc_config $name',
+      '',
+      ': ${' + this.rcServiceName + '_user:="' + this.hbService.asUser + '"}',
+      ': ${' + this.rcServiceName + '_enable:="NO"}',
+      ': ${' + this.rcServiceName + '_facility:="daemon"}',
+      ': ${' + this.rcServiceName + '_priority:="debug"}',
+      ': ${' + this.rcServiceName + '_storage_path:="' + this.hbService.storagePath + '"}',
+      '',
+      'command="/usr/local/bin/${name}"',
```

Comment:
This should be `this.hbService.selfPath`; I think it's being set to `/usr/local/bin/homebridge` if I'm reading this correctly?

The `hb-service run ...` command should be used instead of launching `homebridge` as the parent process.

Reply:
selfPath is more correct, yes.

hb-service install uses start(). service homebridge start/stop works. However, the homebridgeBinary path is coming up null, despite homebridge being installed with -g --unsafe-perm. run doesn't work any better for me.